lib/openstack_query/query_blocks/query_executer.py

```diff
@@ -79,6 +84,9 @@ def parse_func(
 
     @property
     def output_func(self):
+        """
+        A getter for output function
+        """
         return self._output_func
```

Comment:
Ditto output what I said about, where the `output_func` should not be in the executor, this should only run the query then return to the `QueryAPI` to marshall onto the class / method that handles output related things

Reply:
refactoring run_parser, and generate_output to be parameters for parse_results instead